torch/nn/functional.py

```diff
@@ -5154,14 +5160,37 @@ def multi_head_attention_forward(
             leads to a significant performance degradation.*
         attn_mask: 2D or 3D mask that prevents attention to certain positions. A 2D mask will be broadcasted for all
             the batches while a 3D mask allows to specify a different mask for the entries of each batch.
-        is_causal: If specified, applies a causal mask as attention mask, and ignores
-            attn_mask for computing scaled dot product attention.
-            Default: ``False``.
-            .. warning::
-                is_causal is provides a hint that the attn_mask is the
-                causal mask.Providing incorrect hints can result in
-                incorrect execution, including forward and backward
-                compatibility.
+
+            is_causal: If ``is_causal=True``, provides a hint that ``mask``
+                is a causal mask which will enable the use of a faster fused
+                ``scaled_dot_product_attention()`` operators (SDPA) on
+                hardware where such enhanced fused SDPA are supported.
+                If ``is_causal=False`, an SDPA operator using the mask will
+                be used (which may result in slower execution).  If
+                ``is_causal=None``, the TransformerEncoder and
+                TransformerDecoder modules may detect whether is_causal os
+                a causal mask. (Detecting the presence of a causal mask
+                imposes some overhead. Thus, specifying ``is_causal=True``
```

Comment:
this last sentence here seems like it's missing something, would it more succinct to phrase this phrased as 
"Thus for best performance the user should specify ``is_causal``."